src/Plugin/Block/SamlLoginBlock.php

```diff
@@ -13,7 +12,7 @@
 use Symfony\Component\HttpFoundation\RequestStack;
 
 /**
- * Provides a 'Saml Login Block' block.
+ * Provides a 'Saml Login/Logout Block' block.
  *
  * @Block(
  *  id = "stanford_samlauth_login_block",
```

Comment:
Since this is a "Login Button", it think it would be better to use a new block plugin.

Reply:
@pookmish Would renaming the block to `SamlLoginLogutBlock` eliminate the need for creating a second block? I believe the original plan was to intentionally keep everything in one block.

Reply:
We can change the name of the file & class but, it's not an option to change the ID of the block plugin. This would cause cascading issues on every platform that uses the module and block.
Because the block id is `login`, it is best to create a `logout` block. you can almost copy the current block plugin and just reverse the access condition and display a logout button instead.